### PR TITLE
add doc deploy and check deadlinks to travis

### PR DESCRIPTION
**Pull request by berthieresteban**
Add doc deploy and dead links check

Boyscout: 

fix deadlinks

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,41 +1,147 @@
...
+        - S3_BUCKET=docs.kuzzle.io
+        - CLOUDFRONT_DISTRIBUTION_ID=E3D6RP0POLCJMM
+  
+    # ----------------------------
+    # Dev deployment
+    # ----------------------------
+    - <<: *deploy-base
+      <<: *doc-deploy-content
+      name: Deploy next-docs.kuzzle.io
+      if: branch = develop
```
Comment (Aschen) [inline .travis.yml]: Do we have a develop branch ? If no, we need to deploy both `docs.kuzzle.io` and `next-docs.kuzzle.io` when pushing on `master`

Comment (alexandrebouthinon) [reply .travis.yml]: He creates one :wink:

File: massive-data-import/README.md [hunk 2]
```diff
@@ -90,7 +90,7 @@ const bulkQuery = {
 }
 ```
 
-This request is then sent to the Bulk Controller with the [Kuzzle.bulk.import](https://docs.kuzzle.io/sdk-reference/js/6/bulk/import/) method from the SDK Javascript.
+This request is then sent to the Bulk Controller with the [Kuzzle.bulk.import](https://docs.kuzzle.io/sdk/js/6/controllers/bulk/import) method from the SDK Javascript.
```
Comment (Shiranuit) [inline massive-data-import/README.md]:
```suggestion
This request is then sent to the Bulk Controller with the [Kuzzle.bulk.import](/sdk/js/6/controllers/bulk/import) method from the SDK Javascript.
```

**Comment by berthieresteban** [comment]
Travis deadlinks tests failed because of /core/1 or /core/2 links are dead. 
But this pr should remove some deadlinks fails and allow other pr to pass the travis deadlinks without how to 404.

File: .travis.yml [hunk 3]
```diff
@@ -1,41 +1,147 @@
...
-  apt:
-    packages:
-      - ruby
-      - awscli
+      install:
+        - pip install awscli --upgrade --user
+      before_install:
+        - sudo sysctl -w vm.max_map_count=262144
+        # Build images
+        - docker login -u aschen -p $DOCKER_PASSWORD
```
Comment (xbill82) [inline .travis.yml]: `aschen` :question:

Comment (Aschen) [inline .travis.yml]:
```suggestion
        - docker login -u kuzzleteam -p $DOCKER_PASSWORD
```

Comment (Aschen) [reply .travis.yml]: And the `DOCKER_PASSWORD` has to be changed

File: .travis.yml [hunk 4]
```diff
@@ -1,41 +1,145 @@
...
-  apt:
-    packages:
-      - ruby
-      - awscli
+      install:
+        - pip install awscli --upgrade --user
+      before_install:
+        - sudo sysctl -w vm.max_map_count=262144
+        # Build images
+        - docker login -u kuzzleteam -p $DOCKER_PASSWORD
```
Comment (Aschen) [inline .travis.yml]: Missing `DOCKER_PASSWORD` variable in secure env variables

File: .travis.yml [hunk 5]
```diff
@@ -1,41 +1,145 @@
...
+# Deployments
+_deploy-base: &deploy-base
+  stage: Deployments
+  language: node_js
+  node_js: 10
+  env: &deploy-base-env
+    - NODE_ENV=production
+    - AWS_DEFAULT_REGION=us-west-2
+    - S3_BUCKET=dl.kuzzle.io
+    - DOCKER_COMPOSE_VERSION=1.12.0
```
Comment (Aschen) [inline .travis.yml]: Could be removed, this variable is not taken into account by Travis

File: .travis.yml (hunk 4 above)
Comment (berthieresteban) [reply .travis.yml]: I've add it into Travis

Comment (Aschen) [reply .travis.yml]: Hmm ok, usually we prefer to put everything in the `.travis.yml` file so there is no hidden var but I won't block this PR for this

**Pull request merged**